terraform/node_resource_abstract.go

```diff
@@ -321,11 +323,17 @@ func (n *NodeAbstractResource) ImpliedProvider() addrs.Provider {
 func (n *NodeAbstractResourceInstance) ProvidedBy() (addrs.ProviderConfig, bool) {
 	// If we have a config we prefer that above all else
 	if n.Config != nil {
-		relAddr := n.Config.ProviderConfigAddr()
-		return addrs.LocalProviderConfig{
-			LocalName: relAddr.LocalName,
-			Alias:     relAddr.Alias,
-		}, false
+		relAddr := n.Config.ProviderConfigRef
```

Comment:
This is where I feel that the change is most obvious, and possibly problematic.
This tripped me up: in the current code, `n.Config.ProviderConfigAddr()` returns the implied default provider local config when the `ProviderConfigRef` is nil. 

As a human reading the code, I like the fact that it's obvious that the result returned by ProvidedBy() could be an implied provider, but it's not the prettiest code and there may be implications I've missed.